Lib/pydoc.py

```diff
@@ -2674,6 +2663,11 @@ def get_html_page(url):
             title, content = html_error(complete_url, exc)
         return html.page(title, content)
 
+    # Make sure the secret token is inside before doing any URL operations.
+    left, sep, right = url.partition(SECRET_URL_TOKEN)
```

Comment:
Oops I just realized this possibly susceptible to timing attacks,

TODO for self: this should use ``secrets.compare_digest``